app/src/main/java/com/ichi2/apisample/MainActivity.java

```diff
@@ -289,9 +383,16 @@ protected void restoreUiState() {
 
     public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
         switch (requestCode) {
+            case AD_PERM_REQUEST_VALID:
             case AD_PERM_REQUEST: {
                 if (grantResults.length > 0 && grantResults[0] != PackageManager.PERMISSION_GRANTED) {
                     showMsg(R.string.anki_permission_denied);
+                } else if (requestCode == AD_PERM_REQUEST_VALID) {
```

Comment:
There is a mix of "switch/case" and "if" operations with the same variable, which looks strange.